miniserver/routes/index.js

```diff
@@ -0,0 +1,140 @@
+        return res.send(response.toString())
+      }
+      // Allow requesting of funds from users by address (without checking registered phone number)
+      if (smsParams[1].substr(0, 2) === '0x') {
+        funder.address = smsParams[1]
+      } else {
+      // set country and look up user address from Phone Number
+        let funderPhone = smsParams[1]
+        if (funderPhone.substr(0, 1) !== '+') {
+          const { countryIso3 } = phone(requestor.phone)
+          const { isValid, phoneNumber } = phone(funderPhone, countryIso3)
+          funderPhone = phoneNumber
+          if (!isValid) {
+            response.message(`error: invalid recipient phone number ${smsParams[1]}. example request "p +14158401999 0.1"`)
+            return res.send(response.toString())
+          }
+        }
+        funder = await User.findByPhone({ phone: funderPhone })
+        if (funder ? !funder.address : true) {
+          response.message(`error: funders phone number is not a registered wallet: ${smsParams[1]}. example request "p +14158401999 0.1"`)
+          return res.send(response.toString())
+        }
+      }
+
+      if (toNumber(smsParams[2]) > 0) {
+        amount = ethers.utils.parseEther(smsParams[2])
+      } else {
+        response.message(`error: pay request requires a valid amount': ${smsParams[2]} example request "p +14158401999 0.1"`)
+        return res.send(response.toString())
+      }
+      req.processedBody = { ...req.processedBody, command, requestor, funder, amount }
+    } else {
+      response.message('error: invalid sms command. example payment request "p +14158401999 0.1"')
+      return res.send(response.toString())
+    }
+  } catch (ex) {
+    console.error(ex)
+    const { code, error, success } = parseError(ex)
+    response.message(`An exception occured. success: ${success} code: ${code} error: ${error.substr(0, 100)} `)
+    return res.send(response.toString())
+  }
+  next()
+}
+
+router.post('/sms', checkfromTwilio, parseSMS, async (req, res) => {
+  const messagingResponse = Twilio.twiml.MessagingResponse
+  const response = new messagingResponse()
+  // Look up the from Phone Number to get the address
+  const { command, requestor, funder, amount } = req.processedBody
+  Logger.log(`req.body: ${JSON.stringify(req.processedBody)}`)
+  const miniWallets = blockchain.getMiniWallets()
+  const miniWallet = miniWallets[1]
+
+  const logger = (...args) => Logger.log('[/sms]', ...args)
+  const executor = blockchain.prepareExecute(logger)
+
+  Logger.log(`miniWallet.address: ${miniWallet.address}`)
+  try {
+    if (command === 'balance') {
+      const balance = ethers.utils.formatEther(await miniWallet.userBalances(requestor.address))
+      const messagingResponse = Twilio.twiml.MessagingResponse
+      const response = new messagingResponse()
+      response.message(`phone: ${requestor.phone} address: ${requestor.address}, balance: ${balance} `)
```

Comment:
Your balance is [balance] (address: [address])

Reply:
Updated 🙏